From f087db5f78d0b5a9ac67f0b664b115ba729c2302 Mon Sep 17 00:00:00 2001
From: Gleb Pinigin <gpinigin@gmail.com>
Date: Fri, 2 Aug 2013 19:41:55 +0700
Subject: [PATCH 16/16] vout_ios2: take into account scale of attached screen

As said in Apple documentation drawRect should not be implemented for view based on opengl es layer.
Instead contentScaleFactor should be changed manually if needed. Underlying opengl es layer will adjust its scale accordingly.
---
 modules/video_output/ios2.m | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/modules/video_output/ios2.m b/modules/video_output/ios2.m
index 28369b4..397be6d 100644
--- a/modules/video_output/ios2.m
+++ b/modules/video_output/ios2.m
@@ -393,10 +393,10 @@ static void OpenglESSwap(vlc_gl_t *gl)
     [super dealloc];
 }
 
-/* we don't get the correct scale factor if we don't overwrite this method */
-- (void)drawRect:(CGRect) rect
+- (void)didMoveToWindow
 {
-    [super drawRect:rect];
+    self.contentScaleFactor = self.window.screen.scale;
+    _bufferNeedReset = YES;
 }
 
 - (void)createBuffers
-- 
1.7.12.4 (Apple Git-37)

